### PR TITLE
Mon 16533 asio feeders

### PR DESCRIPTION
**Pull request by jean-christophe81**
## Description

Before this evolution, failovers create one thread per incomming connection. Then it uses asio do do the job

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [X] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [X] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: tests/resources/resources.robot [hunk 1]
```diff
@@ -260,6 +260,13 @@ Coredump Info
     ...    stdout=${output}
     ...    stderr=${output}
 
+Copy Coredump In Failed Dir
+    [Arguments]    ${process_name}    ${binary_path}    ${name}
+    ${pid}=    Get Process Id    ${process_name}
+    ${failDir}=    Catenate    SEPARATOR=    failed/    ${Test Name}
```
Comment (thearchit3ct) [inline tests/resources/resources.robot:266]:
```suggestion
    ${pid}    Get Process Id    ${process_name}
    ${failDir}    Catenate    SEPARATOR=    failed/    ${Test Name}
```

**Review by bouda1 — APPROVED** [review]

File: tests/broker-engine/services-and-bulk-stmt.robot [hunk 2]
```diff
@@ -67,25 +69,33 @@ EBBPS1
     FOR    ${i}    IN RANGE    ${1000}
         Process Service Check result    host_1    service_${i+1}    2    warning${i}
         IF    ${i} % 200 == 0
+            ${first_service_status_content}    Create List    unified_sql service_status processing
+            ${result}    Find In Log with timeout
```
Comment (thearchit3ct) [inline tests/broker-engine/services-and-bulk-stmt.robot:73]:
```suggestion
            ${result}    Find In Log With Timeout
```

File: tests/broker-engine/services-and-bulk-stmt.robot [hunk 3]
```diff
@@ -67,25 +69,33 @@ EBBPS1
...
             Start Broker
         END
     END
-    ${content}=    Create List
+    ${content}    Create List
     ...    connected to 'MariaDB' Server
     ...    it supports column-wise binding in prepared statements
-    ${result}=    Find In Log with timeout    ${centralLog}    ${start}    ${content}    30
-    Should Be True    ${result}    msg=Prepared statements should be supported with this version of MariaDB.
+    ${result}    Find In Log with timeout    ${centralLog}    ${start}    ${content}    30
```
Comment (thearchit3ct) [inline tests/broker-engine/services-and-bulk-stmt.robot:91]:
```suggestion
    ${result}    Find In Log With Timeout    ${centralLog}    ${start}    ${content}    30
```

File: tests/broker-engine/services-and-bulk-stmt.robot [hunk 4]
```diff
@@ -105,33 +115,35 @@ EBBPS2
...
     FOR    ${i}    IN RANGE    ${1000}
         Process Service Check result    host_1    service_${i+1}    1    warning${i}
     END
-    ${content}=    Create List
+    ${content}    Create List
     ...    connected to 'MariaDB' Server
     ...    it supports column-wise binding in prepared statements
-    ${result}=    Find In Log with timeout    ${centralLog}    ${start}    ${content}    30
-    Should Be True    ${result}    msg=Prepared statements should be supported with this version of MariaDB.
+    ${result}    Find In Log with timeout    ${centralLog}    ${start}    ${content}    30
```
Comment (thearchit3ct) [inline tests/broker-engine/services-and-bulk-stmt.robot:139]:
```suggestion
    ${result}    Find In Log With Timeout    ${centralLog}    ${start}    ${content}    30
```

File: tests/broker-engine/services-and-bulk-stmt.robot [hunk 5]
```diff
@@ -140,27 +152,34 @@ EBBPS2
     Should Be Equal As Strings    ${output}    ((0,),)
 
     FOR    ${i}    IN RANGE    ${1000}
-        Process Service Check result    host_1    service_${i+1}    2    warning${i}
+        Process Service Check result    host_1    service_${i+1}    2    critical${i}
```
Comment (thearchit3ct) [inline tests/broker-engine/services-and-bulk-stmt.robot:155]:
```suggestion
        Process Service Check Result    host_1    service_${i+1}    2    critical${i}
```

File: tests/broker-engine/services-and-bulk-stmt.robot [hunk 6]
```diff
@@ -140,27 +152,34 @@ EBBPS2
...
         IF    ${i} % 200 == 0
-            Log to Console    Stopping Broker
+            ${first_service_status_content}    Create List    unified_sql service_status processing
+            ${result}    Find In Log with timeout
```
Comment (thearchit3ct) [inline tests/broker-engine/services-and-bulk-stmt.robot:158]:
```suggestion
            ${result}    Find In Log With Timeout
```

File: tests/broker-engine/services-and-bulk-stmt.robot [hunk 7]
```diff
@@ -140,27 +152,34 @@ EBBPS2
...
             Start Broker
         END
     END
-    ${content}=    Create List
+    ${content}    Create List
     ...    connected to 'MariaDB' Server
     ...    it supports column-wise binding in prepared statements
-    ${result}=    Find In Log with timeout    ${centralLog}    ${start}    ${content}    30
-    Should Be True    ${result}    msg=Prepared statements should be supported with this version of MariaDB.
+    ${result}    Find In Log with timeout    ${centralLog}    ${start}    ${content}    30
```
Comment (thearchit3ct) [inline tests/broker-engine/services-and-bulk-stmt.robot:175]:
```suggestion
    ${result}    Find In Log With Timeout    ${centralLog}    ${start}    ${content}    30
```

File: tests/broker-engine/services-and-bulk-stmt.robot [hunk 8]
```diff
@@ -186,29 +205,29 @@ EBMSSM
...
     Start Broker
     Start Engine
     Broker Set Sql Manager Stats    51001    5    5
 
     # Let's wait for the external command check start
-    ${content}=    Create List    check_for_external_commands()
-    ${result}=    Find In Log with Timeout    ${engineLog0}    ${start}    ${content}    60
-    Should Be True    ${result}    msg=A message telling check_for_external_commands() should be available.
+    ${content}    Create List    check_for_external_commands()
+    ${result}    Find In Log with Timeout    ${engineLog0}    ${start}    ${content}    60
```
Comment (thearchit3ct) [inline tests/broker-engine/services-and-bulk-stmt.robot:215]:
```suggestion
    ${result}    Find In Log With Timeout    ${engineLog0}    ${start}    ${content}    60
```

File: tests/broker-engine/services-and-bulk-stmt.robot [hunk 9]
```diff
@@ -186,29 +205,29 @@ EBMSSM
...
+    Should Be True    ${result}    A message telling check_for_external_commands() should be available.
 
-    ${start}=    Get Round Current Date
+    ${start}    Get Round Current Date
     # Let's wait for one "INSERT INTO data_bin" to appear in stats.
     FOR    ${i}    IN RANGE    ${1000}
         Process Service Check result with metrics    host_1    service_${i+1}    1    warning${i}    100
```
Comment (thearchit3ct) [inline tests/broker-engine/services-and-bulk-stmt.robot:221]:
```suggestion
        Process Service Check Result With Metrics    host_1    service_${i+1}    1    warning${i}    100
```

File: tests/broker-engine/services-and-bulk-stmt.robot [hunk 10]
```diff
@@ -236,22 +255,22 @@ EBPS2
...
-    Start Broker    
+    ${start}    Get Current Date
+    Start Broker
     Start Engine
     # Let's wait for the external command check start
-    ${content}=    Create List    check_for_external_commands()
-    ${result}=    Find In Log with Timeout    ${engineLog0}    ${start}    ${content}    60
-    Should Be True    ${result}    msg=A message telling check_for_external_commands() should be available.
+    ${content}    Create List    check_for_external_commands()
+    ${result}    Find In Log with Timeout    ${engineLog0}    ${start}    ${content}    60
```
Comment (thearchit3ct) [inline tests/broker-engine/services-and-bulk-stmt.robot:263]:
```suggestion
    ${result}    Find In Log With Timeout    ${engineLog0}    ${start}    ${content}    60
```

File: tests/broker-engine/services-and-bulk-stmt.robot [hunk 11]
```diff
@@ -236,22 +255,22 @@ EBPS2
...
     FOR    ${i}    IN RANGE    ${1000}
         Process Service Check result with metrics    host_1    service_${i+1}    1    warning${i}    20
     END
-    ${start}=    Get Current Date
-    ${content}=    create list    Check if some statements are ready,    sscr_bind connections
-    ${result}=    Find In Log with Timeout    ${centralLog}    ${start}    ${content}    60
-    Should Be True    ${result}    msg=A message telling that statements are available should be displayed
+    ${start}    Get Current Date
+    ${content}    create list    Check if some statements are ready,    sscr_bind connections
+    ${result}    Find In Log with Timeout    ${centralLog}    ${start}    ${content}    60
```
Comment (thearchit3ct) [inline tests/broker-engine/services-and-bulk-stmt.robot:272]:
```suggestion
    ${content}    Create list    Check if some statements are ready,    sscr_bind connections
    ${result}    Find In Log With Timeout    ${centralLog}    ${start}    ${content}    60
```

File: tests/broker-engine/services-and-bulk-stmt.robot [hunk 12]
```diff
@@ -275,58 +294,55 @@ RLCode
...
 
     # Start the engine/broker
-    ${start}=    Get Current Date
+    ${start}    Get Current Date
 
     Start Broker
     Start Engine
 
+    ${content}    Create List    check_for_external_commands()
+    ${result}    Find In Log with Timeout    ${engineLog0}    ${start}    ${content}    60
```
Comment (thearchit3ct) [inline tests/broker-engine/services-and-bulk-stmt.robot:319]:
```suggestion
    ${result}    Find In Log With Timeout    ${engineLog0}    ${start}    ${content}    60
```

File: tests/broker-engine/services-and-bulk-stmt.robot [hunk 13]
```diff
@@ -30,33 +30,35 @@ EBBPS1
...
     FOR    ${i}    IN RANGE    ${1000}
         Process Service Check result    host_1    service_${i+1}    1    warning${i}
     END
-    ${content}=    Create List
+    ${content}    Create List
     ...    connected to 'MariaDB' Server
     ...    it supports column-wise binding in prepared statements
-    ${result}=    Find In Log with timeout    ${centralLog}    ${start}    ${content}    30
-    Should Be True    ${result}    msg=Prepared statements should be supported with this version of MariaDB.
+    ${result}    Find In Log with timeout    ${centralLog}    ${start}    ${content}    30
```
Comment (thearchit3ct) [inline tests/broker-engine/services-and-bulk-stmt.robot:54]:
```suggestion
    ${result}    Find In Log With Timeout    ${centralLog}    ${start}    ${content}    30
```

File: tests/resources/Broker.py [hunk 14]
```diff
@@ -1715,6 +1715,30 @@ def compare_rrd_average_value(metric, value: float):
...
+        expected = 100
+        if (value == 1):
+            expected = 75
+        elif(value == 2):
+            expected = 0
+        return expected == res
+    else:
+        logger.console(
+            f"It was impossible to get the average value from the file {VAR_ROOT}/lib/centreon/statuss/{index_id}.rrd from the last 30 days")
+        return True
```
Comment (thearchit3ct) [inline tests/resources/Broker.py:1739]:
```suggestion
    res = getoutput(f"rrdtool graph dummy --start=end-180d --end=now DEF:x={VAR_ROOT}/lib/centreon/status/{index_id}.rrd:value:AVERAGE VDEF:xa=x,AVERAGE PRINT:xa:%lf")
    lst = res.split('\n')
    if len(lst) >= 2:
        res = float(lst[1].replace(',', '.'))
        expected = {1: 75, 2: 0}.get(value, 100)
        return expected == res
    else:
        logger.console(f"It was impossible to get the average value from the file {VAR_ROOT}/lib/centreon/statuss/{index_id}.rrd from the last 30 days")
        return True
```

**Review by thearchit3ct — APPROVED** [review]

**Review by bouda1 — APPROVED** [review]
Broker.py at line 1723 should be simplified...

**Comment by jean-christophe81** [comment]
REFS:MON-21484

**Pull request merged**